File: src/main/java/com/example/treemenu/model/Menu.java
package com.example.treemenu.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "test_categories")
public class Menu {
    @Id
    private Integer id;
    @Column(name = "parent_id")
    private Integer pid;
    @Column
    private String name;


    public Menu(){
    }

    public Menu(String name, Integer id, Integer pid) {
        this.id = id;
        this.pid = pid;
        this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getPid() {
        return pid;
    }

    public void setPid(Integer pid) {
        this.pid = pid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: src/main/kotlin/com/example/treemenu/model/Menu.kt
package com.example.treemenu.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "test_categories")
class Menu {
    @Id
    var id: Int? = null

    @Column(name = "parent_id")
    var pid: Int? = null

    @Column
    var name: String? = null
}